library/src/main/java/com/google/android/exoplayer2/text/eia608/Eia608Decoder.java

```diff
@@ -463,19 +719,19 @@ private String getDisplayCaption() {
 
     int endIndex = endsWithNewline ? buildLength - 1 : buildLength;
     if (captionMode != CC_MODE_ROLL_UP) {
-      return captionStringBuilder.substring(0, endIndex);
+      return captionStringBuilder.subSequence(0, endIndex);
     }
 
     int startIndex = 0;
     int searchBackwardFromIndex = endIndex;
     for (int i = 0; i < captionRowCount && searchBackwardFromIndex != -1; i++) {
-      searchBackwardFromIndex = captionStringBuilder.lastIndexOf("\n", searchBackwardFromIndex - 1);
+      searchBackwardFromIndex = captionStringBuilder.toString().lastIndexOf("\n", searchBackwardFromIndex - 1);
```

Comment:
[SpannableStringBuilder](https://developer.android.com/reference/android/text/SpannableStringBuilder.html#delete(int, int)) does not offer the lastIndexOf method.

And yes, most likely it can be removed after we figured out: https://github.com/google/ExoPlayer/pull/1773#discussion_r76045863